### PR TITLE
Configure client with NoAuthAuthScheme by default

### PR DESCRIPTION
**Pull request by sugmanue**
## Motivation and Context

This change adds the `NoAuthAuthScheme` to the list of available auth schemes to all services.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Review by cenedhryn — APPROVED** [review]

File: codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeSpecUtils.java [hunk 1]
```diff
@@ -197,6 +198,8 @@ public Set<Class<?>> allServiceConcreteAuthSchemeClasses() {
             // sigv4a is not modeled but needed for the endpoints based auth-scheme cases.
             result.add(AwsV4aAuthScheme.class);
         }
+        // Make the no-auth scheme available.
+        result.add(NoAuthAuthScheme.class);
```
Comment (gosar) [inline codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeSpecUtils.java:202]: not your change, but looking at where this method is used, looks like there is a potential bug in BaseClientBuilderClass.authSchemesMethod() that would have returned emptyMap before your change, ignoring the additionalAuthSchemes. I think this PR makes it irrelevant (as the collection will never be null). I was thinking we are missing a test case for service with zero auth schemes, but looks like C2J just doesn't allow it today. Either ways, with your PR, looks like 
```
        if (concreteAuthSchemeClasses.isEmpty()) {
            builder.addStatement("return $T.emptyMap()", Collections.class);
            return builder.build();
        }
```
can be removed. I have done it locally, will send PR

Comment (gosar) [reply codegen/src/main/java/software/amazon/awssdk/codegen/poet/auth/scheme/AuthSchemeSpecUtils.java:202]: https://github.com/aws/aws-sdk-java-v2/pull/4364 for this
